### PR TITLE
Fixes #23767: Make an install subcommand for the new rudder-package

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23767

### CODE REVIEW

File: relay/sources/rudder-package/src/archive.rs [hunk 1]
```diff
@@ -184,11 +188,22 @@ impl Rpkg {
         Ok(current_database.is_installed(self.to_owned()))
     }
 
-    pub fn install(&self) -> Result<()> {
-        let keys = self.metadata.content.keys().clone();
+    pub fn install(&self, force: bool) -> Result<()> {
+        debug!("Trying to install rpkg '{}'...", self.path);
```
Comment (amousset) [inline relay/sources/rudder-package/src/archive.rs:192]:
```suggestion
        debug!("Installing rpkg '{}'...", self.path);
```

File: relay/sources/rudder-package/src/archive.rs [hunk 2]
```diff
@@ -184,11 +188,22 @@ impl Rpkg {
...
+        // Verify webapp compatibility
+        let webapp_version = RudderVersion::from_path(RUDDER_VERSION_FILE)?;
+        if !(force
+            || self
+                .metadata
+                .version
+                .rudder_version
+                .is_compatible(&webapp_version.to_string()))
+        {
+            bail!("This plugin was build for a webapp version '{}', it is incompatible with your current webapp version '{}'.", self.metadata.version.rudder_version, webapp_version)
```
Comment (amousset) [inline relay/sources/rudder-package/src/archive.rs:202]:
```suggestion
            bail!("This plugin was built for Rudder '{}', it is incompatible with your current version '{}'.", self.metadata.version.rudder_version, webapp_version)
```

File: relay/sources/rudder-package/src/archive.rs [hunk 3]
```diff
@@ -226,19 +243,34 @@ impl Rpkg {
...
             }
         }
+        // Restarting webapp
+        debug!("Install completed");
         Ok(())
     }
 
     fn run_package_script(&self, script: PackageScript, arg: PackageScriptArg) -> Result<()> {
+        debug!(
+            "Trying to run package script '{}' with args '{}' for rpkg '{}'...",
```
Comment (amousset) [inline relay/sources/rudder-package/src/archive.rs:253]:
```suggestion
            "Running package script '{}' with args '{}' for rpkg '{}'...",
```

File: relay/sources/rudder-package/src/cli.rs [hunk 4]
```diff
@@ -15,4 +15,19 @@ pub struct Args {
...
+}
+
+#[derive(Debug, Subcommand)]
+pub enum Command {
+    Install {
+        #[clap(long, short = 'f', help = "Force installation of given plugin")]
+        force: bool,
+
+        #[clap()]
+        package: Option<String>,
```
Comment (amousset) [inline relay/sources/rudder-package/src/cli.rs:30]: why an Option?

Comment (Fdall) [reply relay/sources/rudder-package/src/cli.rs:30]: Because of the `--list` option that's been removed. Will fix

**Comment by Fdall** [comment]
PR updated with a new commit

File: relay/sources/rudder-package/src/lib.rs [hunk 5]
```diff
@@ -47,10 +49,42 @@ pub fn run() -> Result<()> {
...
+        } else {
+            bail!("TODO");
+        };
+        let rpkg = Rpkg::from_path(&rpkg_path)?;
+        rpkg.install(force)
+    }
+
+    pub fn list() -> Result<()> {
+        let db = Database::read(PACKAGES_DATABASE_PATH);
+        info!("Installed plugins:\n{:?}", db);
```
Comment (amousset) [inline relay/sources/rudder-package/src/lib.rs:87]: I think it as a println and not a log

Comment (amousset) [reply relay/sources/rudder-package/src/lib.rs:87]: we want to dsiplay it regardless of the log level

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**